### DISCUSSION TITLE
Issue with deployment of SunbirdEdportal version 4.10.0 deployment

### DISCUSSION THREAD

**Discussion by paritshivani** [Sunbird-Ed/Community#578] [Installation]
Hi folks!
hope you are doing great.

We are upgrading our project from version 4.4 to version 4.10.0. We used tag 4.10.0_RC12. We build this branch in april and deployed it was successful and working fine.
Now again We used same tag and tried to build we are got following error

![image (4)](https://github.com/Sunbird-Ed/Community/assets/58217306/595e15d3-7302-4304-aa0e-1968bfb761ad)

Then we upgraded the node verion to 16 to resolve build and deployment issues. With this, build and deployment got successful but now we are facing another issue

After deployment, few APIs are taking to long to exicute and giving timeout error, and henceforth site is giving error as 504 gateway timeout.

If we use previously working build (from april) for deployment it works all good again. 

we need urgent help in this. How we can solve this????
@coolbung, @vinukumar-vs 
cc: @Renuka-S

**Comment by chitranshu-keshav** [comment]
@vinukumar-vs @rajeevsathish @gandham-santhosh Can you please get back on this

**Reply by princegupta1131** [reply]
Hi, @paritshivani could you please share the API details & error which taking too long to execute and giving a timeout error?

CC:@vinukumar-vs @rajeevsathish @gandham-santhosh

**Comment by paritshivani** [comment]
@princegupta1131 , there no perticular list of APIs, Everytimt We login some different Apis takes time and give timeout error

**Comment by rajeevsathish** [comment]
@paritshivani we are also using the new version of node and also some of the adopters also have updated the same and has no issue and node version update should not effect any API so please give more details as what is the API and logs for the API so that we can see the issue in detail. 
cc: @chitranshu-keshav , @princegupta1131

**Reply by gandham-santhosh** [reply]
Looks like issue is due to the way dns lookup is changed from 16.7.0 node version. 

https://github.com/chimurai/http-proxy-middleware/issues/705
https://raw.githubusercontent.com/nodejs/node/main/doc/changelogs/CHANGELOG_V16.md 

@rajeevsathish  @princegupta1131 can you please check this?

**Reply by paritshivani** [reply]
@rajeevsathish , Can you please help me with the following things?
1. which version of node you have used with portal version 4.10.0
2. As you mentioned that you have updated some adopters, Can we have a list of that adopters to confirm on our side?

**Comment by paritshivani** [comment]
Findings till now:
1. We have installed on dependency named as : mathjax-full : https://www.npmjs.com/package/mathjax-full
2. This library as internal dependency on : speech-rule-engine : https://www.npmjs.com/package/speech-rule-engine?activeTab=code
3. Speech-rule-engine have internal dependency on package commander@>=7.0.0
4. Till last month commander was having latest version as 10.0.0 which used to support node 14.19.0 but the latest commander@11 needs node 16.0.0
5. If we move to node 16.0.0 we can successfully deploy the portal but face 504 gateway timeout error

Is there are any Solution for this???

**Reply by rajeevsathish** [reply]
@paritshivani even I don't have a solution for this what is the API that is failing if you can give me that I can have a look what can be the issue

@gandham-santhosh / @santhosh-tg  can you please have a look any api is failing in staging with 504 error and if so can you please give me the list so that I can check for a solution.

**Reply by paritshivani** [reply]
Apis are random in this case 
Sharing few below
`"Request URL:
https://host/resourcebundles/v1/read/en
Request Method:
GET
Status Code:
504 Gateway Time-out
Remote Address:
20.207.84.251:443
Referrer Policy:
strict-origin-when-cross-origin"`

`"Request URL:
https://host/learner/user/v1/managed/f816a8b7-8752-441e-977b-7d04a44aaeec?sortBy=createdDate&order=desc
Request Method:
GET
Status Code:
504 Gateway Time-out
Remote Address:
20.207.84.251:443
Referrer Policy:
strict-origin-when-cross-origin"`


`"Request URL: https://host/resourcebundles/v1/read/en Request Method: 
GET Status Code: 504 Gateway Time-out Remote Address: 20.207.84.251:443 Referrer Policy: strict-origin-when-cross-origin"`

every time we are having this timeout to new apis

**Comment by paritshivani** [comment]
@rajeevsathish ,
As per our discussion, We have compared package.json from both 4.10 and 6.0 version of sunbird-ed portal app.
We found no significant change other than node npm versions declared in engines

we tried deploying version 6 portal code but we are facing issue there
Is there any other way we can try?

cc: @coolbung, @Renuka-S

**Reply by Krishnaj20** [reply]
@rajeevsathish @gandham-santhosh - who is looking into this issue?

**Reply by gandham-santhosh** [reply]
@rajeevsathish can you please check this?

**Reply by rajeevsathish** [reply]
@paritshivani as release-6.0.0 in staging is working fine without any error or the api being failed please check the env and also the build that you are doing as in release-6.0.0 we have changes the build node version please check and comeback on this
cc: @gandham-santhosh @chitranshu-keshav @Krishnaj20

**Reply by Krishnaj20** [reply]
@vinukumar-vs - for your attention. Can you suggest ways to help the user who is on an unsupported ED version.

**Reply by vinukumar-vs** [reply]
@paritshivani We are also facing some issues with Node-16(502 timeout errors). We will get back with the reason & what is the solution for this issue.
The issue is not related to release versions of EB(4.10 or 6.0). 
Any other issues related release-4.10 we may not able to help much as it is not in support version.

@rajeevsathish @Krishnaj20 @chitranshu-keshav @santhosh-tg

**Reply by chitranshu-keshav** [reply]
@vinukumar-vs is there any update on the Node-16(502 timeout errors)? If the issue is still open, can you share the JIRA ticket so that this thread can be closed
cc @ashokreddy1208

**Reply by chitranshu-keshav** [reply]
@vinukumar-vs awaiting inputs from you
cc @ashokreddy1208
